### ISSUE TITLE
autoTenantCreation not working - object insert returns tenant not found

### ISSUE THREAD

**Issue by owieboy** [weaviate/weaviate#4931]
### How to reproduce this bug?

Create/update a collection and set the following properties

```json
{
...
   "multiTenancyConfig": {
        "enabled": true,
        "autoTenantCreation": true
      },
...
}
```

Verify property was successfully set:
GET http://localhost:8080/v1/schema/TestSchema

```json
...
    "multiTenancyConfig": {
        "autoTenantCreation": true,
        "enabled": true
    },
...
```

Using REST API, insert this test object and set `tenant` with a value for tenant name that has not been (manually) created yet:
POST http://localhost:8080/v1/objects
```json
{
  "class": "TestSchema",
  "properties": {
    "account_id": "1"
  },
  "vector": [
    1
  ],
  "tenant": "TEST1234"
}
```

### What is the expected behavior?

- Tenant `TEST1234` to be created
- Test object inserted to the tenant

### What is the actual behavior?

Result:
```
{
    "error": [
        {
            "message": "put object: import into index nsobject: determine shard: tenant not found: \"TEST1234\""
        }
    ]
}
```

### Supporting information

![image](https://github.com/weaviate/weaviate/assets/7962021/c940e8cb-43e1-4078-b384-11ac35f73406)

### Server Version

1.25.0-rc.0

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by parkerduckworth** [comment]
Hey @owieboy thanks for your report!

Auto tenant creation is actually only available during batch importing.

This wasn't super clear in the docs, and we have since made a [change to clarify this](https://weaviate.io/developers/weaviate/manage-data/multi-tenancy#automatically-add-new-tenants).

Additionally, I've improved the error message that occurs in this case (multi-tenancy and autoTenantCreation are enabled, and a single object insert is attempted with a non-existent tenant) here with #4948. This will be included in the next patch release, 1.25.1.

Apologies for the miscommunication on our part!

**Comment by owieboy** [comment]
Thank you for the the clarifications. I can confirm that the tenant creation and insertion indeed work when using batch operation. 

POST http://localhost:{{WEAVIATE_PORT}}/v1/batch/objects
```json
{
  "fields": [
    "ALL"
  ],
  "objects": [
    {
    "class": "TestSchema",
    "properties": {
        "account_id": "1"
    },
    "vector": [
        1
    ],
    "tenant": "TSTDRV1234"
    }
  ]
}
```

response:
```json
[
    {
        "class": "TestSchema",
        "creationTimeUnix": 1715893985742,
        "id": "99b56cb4-9f7c-49af-9f75-d047d74c05df",
        "lastUpdateTimeUnix": 1715893985742,
        "properties": {
            "account_id": "1"
        },
        "tenant": "TSTDRV12344",
        "vector": [
            1
        ],
        "deprecations": null,
        "result": {
            "status": "SUCCESS"
        }
    }
]
```
